packages/cipherstash-proxy-integration/src/multitenant/set_keyset_name.rs

```diff
@@ -0,0 +1,374 @@
+#[cfg(test)]
+mod tests {
+    use crate::common::{
+        clear, connect_with_tls, random_id, rows_to_vec, simple_query_with_client, trace, PROXY,
+    };
+
+    ///
+    /// Tests that the `SET CIPHERSTASH.KEYSET_NAME = 'keyset_name'` command is handled correctly
+    /// Test both extended and simple query protocols
+    ///
+    #[tokio::test]
+    async fn set_keyset_name_with_extended_query() {
+        trace();
+        clear().await;
+
+        let tenant_keyset_name_1 = std::env::var("CS_TENANT_KEYSET_NAME_1").unwrap();
+        let tenant_keyset_name_2 = std::env::var("CS_TENANT_KEYSET_NAME_2").unwrap();
+
+        let insert_sql = "INSERT INTO encrypted (id, encrypted_int4) VALUES ($1, $2)";
+        let select_sql = "SELECT encrypted_int4 FROM encrypted WHERE id = $1";
+
+        // KEYSET_NAME IS SCOPED TO A CONNECTION
+        // The same client/connection is used for tests
+        let client = connect_with_tls(PROXY).await;
+
+        // DEFAULT_KEYSET_ID SHOULD BE DISABLED FOR THIS TEST
+        // SET KEYSET IS REQUIRED
+        let result = client.query(select_sql, &[&42]).await;
+        assert!(result.is_err());
+
+        //  --------
+        // INSERT and SELECT as TENANT 1
+        // SET tenant 1
+        let sql = format!("SET CIPHERSTASH.KEYSET_NAME = '{tenant_keyset_name_1}'");
+        let result = client.query(&sql, &[]).await;
+        assert!(result.is_ok(), "Setting keyset_name should succeed");
+
+        // INSERT
+        let tenant_1_id = random_id();
+        let encrypted_val = 2;
```

Comment:
nit: I found this name slightly confusing, because I thought '2 looks like a plaintext value', but I suppose this is named like this as it's 'value to be encrypted'